examples/saas-clients/amazons3/working-with-amazons3-client/README.md

```diff
@@ -0,0 +1,422 @@
+# Working with Amazon S3 client
+
+## Overview
+The Ballerina Amazon S3 client provides connectivity to Amazon Simple Storage Service API. It allows users to interact 
+with Amazon S3 API to create, store and download and use data with other services. 
+
+This example explains how to use the S3 client to connect with Amazon S3 instance and perform the following operations:
+* Create Bucket
+* Create Object
+* Get Object
+* Delete Object
+* Delete Bucket
+
+## Prerequisites
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- A Text Editor or an IDE 
+> **Tip**: For a better development experience, install one of the following Ballerina IDE plugins: [VSCode](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina), [IntelliJ IDEA](https://plugins.jetbrains.com/plugin/9520-ballerina)
+- Obtain the credentials such as Access Key, Secret Access Key (API Secret) from Amazon S3 by following the steps below:
+    1. Create an amazon account by visiting <https://aws.amazon.com/s3/>
+    2. Create a new access key, which includes a new secret access key.
+        - To create a new secret access key for your root account, use the [security credentials](https://console.aws.amazon.com/iam/home?#security_credential) page. Expand the Access Keys section, and then click Create New Root Key.
+        - To create a new secret access key for an IAM user, open the [IAM console](https://console.aws.amazon.com/iam/home?region=us-east-1#home). Click Users in the Details pane, click the appropriate IAM user, and then click Create Access Key on the Security Credentials tab.
+    3. Download the newly created credentials, when prompted to do so in the key creation wizard.
+
+Let's get started with a simple Ballerina program to integrate with Amazon S3.  
+
+### Implementation
```

Comment:
Can we come up with a rough diagram on what we build here?

Reply:
Added a diagram.